yt_dlp/extractor/dw.py

```diff
@@ -81,68 +94,60 @@ class DWIE(InfoExtractor):
     }]
 
     def _entries(self, url, graph_api, media_type):
-        if path := {
-            'a': 'videos',
-            'live': ('posts', ..., 'videos'),
-            'program': 'moreContentsFromUnifiedProgram',
-        }.get(media_type):
-            for dct in traverse_obj(graph_api, (
-                *variadic(path), lambda _, v: v['namedUrl'] not in url,
-            )):
-                yield self.url_result(
-                    f'https://www.dw.com{dct["namedUrl"]}', DWIE)
+        path = self._ENTRIES_PATH_MAP[media_type]
+        for dct in traverse_obj(graph_api, (
+            *variadic(path), lambda _, v: v['namedUrl'] not in url,
+        )):
+            yield self.url_result(
+                f'https://www.dw.com{dct["namedUrl"]}', DWIE)
```

Comment:
seems like a job for `utils.urljoin`
```suggestion
            yield self.url_result(
                urljoin('https://www.dw.com/', dct['namedUrl']), DWIE)
```